### ISSUE TITLE
API Issue error and errorCode reply expected but not returned

### ISSUE THREAD

**Issue by kantmakm** [google/exposure-notifications-verification-server#1663]
### TL;DR
Calling the /issue endpoint previously returned a response that included an error key and errorCode code.  Neither of those are being returned in the current release.

**Expected behavior**
Expected to read an `error` and an `errorCode` in server response to an /issue post

**Observed behavior**

When programmatically posting to the /issue endpoint this is the complete server response:
`{"padding":"fi3K3+k65KXhiBettPm2IdG+SOGGzQ5gVc24cneYe3EwxeHS6ADFdLu/+2lKuWB3Pr1anIHWxqQTb3TSIKvfDJLIrnh7/FBbwUVBg2fXedI0FA42DdZ32YvXfZ92i3WayM6/efymLsLEG7rQiF50Pz0t2Pr/ZIwLxPr3Madg2pnSxAyPfwZALk/REr0sJCFGcCJHOoRJZIdgwigZrbFKGZi4ZtKX5bXDV9KCGuwl91Dz5UybE+LtjLBW71tSnEBNlUL8t1WPdvRkuMV/PXKHnLXV4sjHgIA++6vylctU/5EMpPLqTQ71J2FV7/BI8Q6xd+VQyxJKKg0+e5evpODfEDynbWvC4jxKrROU/2M3f5sdcYQVqYfaLisKiKxhtrJQLTdAEgvOl1bv7v+HnSGTg2f5+sz0jRqG/SED72WBsH8QDebMPa4hfEnGuFIqid+DDkxzxFj7nhYQzbo6OpfSIejQvmcYt+tC7Ncp7+0FB+3c/VtT5S5+EDRvRlImO1ijIhyDTYfSDacdPCo1NaI2gl05B3IfPKvsZpADTnpvLxY5aXzp2KnzZEaR4E5esk8Lmwrul0rflxQb9sVVEg5g58ldt4srJA+lKlusWS/hHxc9hQe0COxnwavmOeXhT65mDaPy0JoIauU+/yj5hfyRZAxK2lHZFAMz4KVEFflNKtvik9ijKpnliD9ggE8Wkeaa+5U7r2lbvBS6ramsXmL62X7OVZPKVi5HiUdlZiFzJxrwJBjlS0j7qbynyoc4Cvr9Pwzevhsf+4h4CIyQmgjmdZOCqB8HULXgPOaNyWxAjAIq37m0C38Hfkdpt5kvARUhS7ChPyt+Adm+qlsUaXlTcDoRSHQyQZT/jkD9vlxs4VvAiNDe4pgAti8zPENhcOZmJuTRrs8ZgwNw6rNEPnoHWDh2UiA9rkMD2cCFxWu+LXX5qnpeaPy/3Es/puSbmti191JdkpaXZ7+QVYxrImVfgsSkYg/Zd4CY73a1MkrzNc6AHqzSH18Ny8M9Mv1rI/8lFVgNP4mCi2yv+32chK+8P6tQxlOeH3zpMn19r5UjVITDBD6cKFR3jM5tLFZBtKOpHG839tCXGTGjFKI7VJKHDUI43dkN+DRRzflIANjoM5GcdwzFPOoeUKdCcPNj+CSc+zhUshL/TAWwHMDgH66kQmt57PvDW0kcjsRNBv4FmgKeubpIMcyFS14l/rYN1OYbp++h9exZQMMv86nAX63Xvrwng3rrohEkIDdYwnQLyR+a769j5hTjPGeL6ZagxFKl0d+cm28OnHEQM7ondY0kwCFyaSp17zxfoaDEI0jwtFRvW+1BuQKRz2VhpS7E1LpJ0WM8sU7+tLWSFDVor6Q+YnuovfoVcp8huhNZbctaJlTLtVVSD1gbnKtgo2KVPMqDK/FxWXYEK7Ybg+cOBEk31QYhw48DiLYjbASfXmUQGu7rU4wmmIaGpJmlXobgsNIdCeF51Qupp5e7OTVyhQQ/mEvi50kzOD8SrtZSEdhNNONJpyraSN3KMhQfN2/KyccMiqbolXyMI69gI9k47ovkD88VYuex9VqjcWqxZwAJYHEV8rR01VNDX0EAd8eJG7rbb7emHE0Gg8SAPCZw05NOkGlSzN0hFfh7PXEAL9DXpHoQO5Rp23IaGjuV7bN0r8on085covV89eILsQOmqpWY20X5BhswC3Vfbh54BuFlMkKcGQAqJqqWi0/oznBKOHpAjMuxpYFoXoQEKgtooAZd62ZJEpmvJoSiXbW7vPUgLqqwQO4JzamvyAsPAP/D30C/xJQizF7sy/om25TsyZ55s0z0SmVTJ/nbZ5DtftfKlgDylIE6j/5E6hfcy282CnDnWoVGEBOSWZ6Lrj2Cfch3WGmro/PDCPBdZrRuS9CrO3LQX9VT0uJEUd7TfiAj+ak9wPLgSrSWfW/gPIoJwsMXjtgGmlaIwmuLf1F3v9rC5ttlm05rbSemyBmoJNAhgOhKBa7DkCqelpqLjOiv/4dOr5Z5TK/G7YJRYU6564UllAjB+pYRWUKmg4P3OZSuV/koPuTTLRek5A0gruDK5fL794W+Aeu0rPfFOhdczk0zZInEu0tCVVswabxC9nWqOL7Q2BuBnViZCyQpovJtrIGVTm90ypr97mPrH58NXANopWAkdIQcnpsvXeOP9o7foA5irq+JP+6vZWip51kVCe50DqFeDmYPy/QOTf/3nEiJRM4CkU6jK1+rXcBZNCLmsdZvtksz/LKNPPKP+LIVwB5KkDPO0IRUVNVZ8AMuVLcRzjc3SF0y4LcqVro9XfZrUDpPdatkjjKuOQOER8L4onCS+cjU5DgPBCp0KJ6uC5wMXC+zrjw876WqkbIMNBI+TG390V1+1viM6j3EKflfinnkCenrwmLLoXMyKAU9v9h75J/j8UQW13QMEY8d0puoE5nz45h3+Bs7tdnBEHcSIz+aIz+kgnTwBZscwmhKJmFeu5o3rF795fYd+tuXrQQ/jlbroJ/uJnWpZ5r3lNo6eMqiumg7l6sd/eUwQ/+rd+r47jt0brNpTHdnXUjApRz4zDh3ykz8k/PKWP0uYdkUPZmstnZmuQxbc9IiEElNtF/6mepxZAa7no3m3RCw0Lp1WDHhPrB1IdQ6wY9qLLsnge61nnNjZbRm8Dph59hWZOadrbRXYgaNw/t7j+LozFuSSnraosvabDWtFEM7pGN3","uuid":"5bd88b3b-7983-4a14-b4a6-ad7a00653c95","code":"51601421","expiresAt":"Fri, 22 Jan 2021 18:21:10 UTC","expiresAtTimestamp":1611339670,"longExpiresAt":"Sat, 23 Jan 2021 18:06:10 UTC","longExpiresAtTimestamp":1611425170}`

Which breaks our processing due to lack of `error` key

```
Traceback (most recent call last):
  File "sendcodes_v2.py", line 203, in <module>
    if response_text['errorCode']:
KeyError: 'errorCode'

```

**Additional information**
I did check the most recent release notes and did not see anything related to this issue.

**Comment by mikehelmick** [comment]
Clients should be looking at response codes - it's not guaranteed that a 200 response will contain the error fields.

**Comment by kantmakm** [comment]
I do not disagree, however, when the keys being returned change without documenting them there is a downstream impact.  The API docs do not indicate that error + errorCode are optional keys - additionally, there was nothing in the release notes this week regarding this change of behavior.  This has delayed code generation by 24h for about 1200 positive cases in our state.

**Comment by mikehelmick** [comment]
Sorry that this caused a delay on your side.

I think the best course of action is for us to clearly document that the error fields will only be included in the response if there is an error.

/assign @whaught

**Comment by kantmakm** [comment]
TY - much appreciated

**Comment by whaught** [comment]
Consumers of the API should work with server maintainers to test against upcoming code releases in advance so that releases can be held or modified if there is a downstream breakage which is not easily remedied by clients. In the case of encv.org, that should be encv-test.org.

**Comment by kantmakm** [comment]
Absolutely, that would be ideal.  Also Developers of the API should have QA regression test against existing use cases and document design changes when deprecating features.  However, we're all trying to do our best with the resources we have available and what we *should* be doing is not always possible.  Thanks for your understanding.

**Comment by whaught** [comment]
I have updated the documentation to better differentiate error and success responses. Similarly error responses will never contain the documented keys which are normally returned on success.

This was not expected as an existing use-case: /issue has never responded with `error` for success and all other APIs properly omit `error` and `errorCode` on success. It was an oversight here that the `errorCode` key was included with no value. However it is also unexpected that any in-use APIs have interface changes which are not coordinated in advance. Apologies again for the delay this caused you.

/close
